File: pi_est.py

import random
from pyspark import SparkContext

NUM_SAMPLES = 1000000

def inside(p):
    x, y = random.random(), random.random()
    return x*x + y*y < 1

if __name__ == "__main__":
    
    sc = SparkContext('local', 'pi_est')

    count = sc.parallelize(xrange(0,NUM_SAMPLES)).filter(inside).count()
    print "Pi estimate: %2.6f \n" %(4 * count / float(NUM_SAMPLES))

File: scala/pi_est.scala
import org.apache.spark.sql.SparkSession
import scala.math.random

object pi_est {

    def inside(p:Int) : Boolean = {
        val x = random * 2 - 1
        val y = random * 2 - 1
        return x * x + y * y <= 1
    }

    def main(args: Array[String]): Unit = {
        val spark = SparkSession.builder.appName("pi_est").getOrCreate()
        val NUM_SAMPLES = 1000000
        val count = spark.sparkContext.parallelize(Range(0, NUM_SAMPLES)).filter(inside).count()
        val pi_est = 4.0 * count / NUM_SAMPLES
        println(s"Pi est = $pi_est")
    }
}